packages/trader/src/AppV2/Components/MarketCategory/market-category.tsx

```diff
@@ -0,0 +1,81 @@
+import React, { useRef, useEffect } from 'react';
+import { MarketGroup } from 'AppV2/Utils/symbol-categories-utils';
+import { Tab, Text, CaptionText } from '@deriv-com/quill-ui';
+import MarketCategoryItem from '../MarketCategoryItem';
+import { ActiveSymbols } from '@deriv/api-types';
+import FavoriteSymbols from '../FavoriteSymbols';
+
+type TMarketCategory = {
+    category: MarketGroup;
+    selectedSymbol: string;
+    setSelectedSymbol: (input: string) => void;
+    setIsOpen: (input: boolean) => void;
+    isOpen: boolean;
+};
+
+const MarketCategory = ({ category, selectedSymbol, setSelectedSymbol, setIsOpen, isOpen }: TMarketCategory) => {
+    const itemRefs = useRef<{ [key: string]: HTMLDivElement | null }>({});
+
+    useEffect(() => {
+        let timer: ReturnType<typeof setTimeout>;
+        if (isOpen && category.market === 'all' && selectedSymbol && itemRefs.current[selectedSymbol]) {
+            timer = setTimeout(() => {
+                itemRefs.current[selectedSymbol]?.scrollIntoView({ block: 'center' });
+            }, 50);
+        }
+        return () => {
+            clearInterval(timer);
```

Comment:
```suggestion
            clearTimeout(timer);
```
the `timer` id should be kept in a ref because it will get reset to undefined (L 20) on every `useEffect` run when either of the dep's change (`isOpen`, `category.market`, or `selectedSymbol`).